static/test/javascripts/spec/common/commercial/membership-messages.spec.js

```diff
@@ -51,57 +53,60 @@ define([
                 alreadyVisited = storage.local.get('gu.alreadyVisited');
-                commercialFeatures.membershipMessages = false;
                 storage.local.set('gu.alreadyVisited', 10);
                 done();
             });
 
             afterEach(function () {
-                commercialFeatures.membershipMessages = showMembershipMessages;
+                commercialFeatures.checkWeCanShowMembershipMessages = showMembershipMessages;
                 storage.local.set('gu.alreadyVisited', alreadyVisited);
             });
 
-            it('should not show any messages even to engaged readers', function () {
-                membershipMessages.init();
-                var message = document.querySelector('.js-site-message.site-message--banner');
-                expect(message).toBeNull();
+            it('should not show any messages even to engaged readers', function (done) {
+                membershipMessages.init().then(function () {
+                    var message = document.querySelector('.js-site-message.site-message--banner');
+                    expect(message).toBeNull();
+                }).then(done);
             });
         });
 
         describe('If user not member', function () {
             beforeEach(function (done) {
-                showMembershipMessages = commercialFeatures.membershipMessages;
+                showMembershipMessages = commercialFeatures.checkWeCanShowMembershipMessages;
                 alreadyVisited = storage.local.get('gu.alreadyVisited');
-                commercialFeatures.membershipMessages = true;
+                commercialFeatures.checkWeCanShowMembershipMessages = Promise.resolve(true);
                 fixtures.render(conf);
                 done();
             });
 
             afterEach(function () {
-                commercialFeatures.membershipMessages = showMembershipMessages;
+                commercialFeatures.checkWeCanShowMembershipMessages = showMembershipMessages;
                 storage.local.set('gu.alreadyVisited', alreadyVisited);
                 fixtures.clean(conf.id);
             });
 
             describe('of the UK edition', function () {
-                it('should show a message to engaged readers', function () {
+                it('should show a message to engaged readers', function (done) {
                     config.page = { edition: 'UK' };
                     storage.local.set('gu.alreadyVisited', 10);
-                    membershipMessages.init();
-                    var message = document.querySelector('.js-site-message.site-message--membership-message-uk');
-                    expect(message).not.toBeNull();
+                    membershipMessages.init().then(function () {
+                        var message = document.querySelector('.js-site-message.site-message--membership-message-uk');
+                        expect(message).not.toBeNull();
```

Comment:
By decoupling rendering the template from injecting it into the page, we could make a lot of our modules synchronous in behaviour. This would be a big improvement; lots of current code seems to labour under promise boilerplate.